tests/conftest.py

```diff
@@ -206,7 +208,8 @@ def upstream_instance(upstream_n_distgit):
     lp = LocalProject(path_or_url=str(u))
 
     ups = Upstream(c, pc, lp)
-    return u, ups
+    yield u, ups
+    chdir(old_cwd)
```

Comment:
Maybe a context manager like [this](https://github.com/user-cont/frambo/blob/master/frambo/utils.py#L60)?